File: Code/app/src/main/java/com/example/myapplication/MainActivity.java
// language: java
package com.example.myapplication;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;

import com.google.firebase.auth.FirebaseAuth;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Toolbar tb = findViewById(R.id.toolbar);
        setSupportActionBar(tb);

        Button btnHome = findViewById(R.id.btn_home);
        Button btnSignOut = findViewById(R.id.btn_sign_out);
        Button btnCreate = findViewById(R.id.btn_create);
        Button btnBrowse = findViewById(R.id.btn_browse);

        // Hide create button for students
        if (UserSession.isStudent()) {
            btnCreate.setVisibility(View.GONE);
        } else {
            btnCreate.setVisibility(View.VISIBLE);
        }

        // Default: show Browse (calendar) fragment
        if (getSupportFragmentManager().findFragmentById(R.id.fragment_container) == null) {
            getSupportFragmentManager().beginTransaction()
                    .replace(R.id.fragment_container, new BrowseFragment())
                    .commit();
        }

        btnBrowse.setOnClickListener(v ->
                getSupportFragmentManager().beginTransaction()
                        .replace(R.id.fragment_container, new BrowseFragment())
                        .addToBackStack(null)
                        .commit()
        );

        btnCreate.setOnClickListener(v -> {
            // open event create if organizer, otherwise do nothing
            if (!UserSession.isStudent()) {
                getSupportFragmentManager().beginTransaction()
                        .replace(R.id.fragment_container, new EventCreateFragment())
                        .addToBackStack(null)
                        .commit();
            }
        });

        btnHome.setOnClickListener(v ->
                getSupportFragmentManager().beginTransaction()
                        .replace(R.id.fragment_container, new BrowseFragment())
                        .addToBackStack(null)
                        .commit()
        );

        btnSignOut.setOnClickListener(v -> {
            FirebaseAuth.getInstance().signOut();
            Intent it = new Intent(MainActivity.this, LoginActivity.class);
            startActivity(it);
            finish();
        });
    }
}

File: Code/app/src/main/java/com/example/eventdetails/MainActivity.kt
package com.example.eventdetails

import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.activity.enableEdgeToEdge
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.Scaffold
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.tooling.preview.Preview
import com.example.eventdetails.ui.theme.EventDetailsTheme
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.FragmentActivity
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.commit
import androidx.fragment.app.replace

class MainActivity : AppCompatActivity () {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // Start with EventListFragment by default
        supportFragmentManager.beginTransaction()
            .replace(R.id.fragment_container, EventListFragment())
            .commit()

        val createBtn = findViewById<Button>(R.id.btnCreate)
        val listBtn = findViewById<Button>(R.id.btnList)

        createBtn.setOnClickListener {
            supportFragmentManager.beginTransaction()
                .replace(R.id.fragment_container, CreateEventFragment())
                .addToBackStack(null)
                .commit()
        }

        listBtn.setOnClickListener {
            supportFragmentManager.beginTransaction()
                .replace(R.id.fragment_container, EventListFragment())
                .commit()
        }
    }
}